kmir/src/kmir/kdist/mir-semantics/rt/value.md

```diff
@@ -33,9 +33,9 @@ The special `Moved` value represents values that have been used and should not b
                    // heterogenous value list        for tuples and structs (standard, tuple, or anonymous)
                  | Float( Float, Int )                    [symbol(Value::Float)]
                    // value, bit-width               for f16-f128
-                 | Reference( Int , Place , Mutability , Metadata )
+                 | Reference( Int , Place , Mutability , Metadata , Int )
                                                           [symbol(Value::Reference)]
-                   // stack depth (initially 0), place, borrow kind, dynamic size if applicable
+                   // stack depth (initially 0), place, borrow kind, dynamic size if applicable, pointer offset (for references made from offset raw pointers)
```

Comment:
After reading through all the K code in `data.md`, I think it probably makes sense to not store the offset separately but to combine it with the `Metadata`. Maybe the `PtrEmulation` should be used in `Reference`, too? For all values that have `noMetadata` it just does not make sense to have an offset `/= 0` .